libs/entitlement/asm-provider/src/test/java/org/elasticsearch/entitlement/instrumentation/impl/InstrumentationServiceImplTests.java

```diff
@@ -71,17 +78,11 @@ public void testInstrumentationTargetLookup() throws IOException, ClassNotFoundE
         assertThat(
             checkMethods,
             hasEntry(
-                equalTo(
-                    new MethodKey(
-                        "org/elasticsearch/entitlement/instrumentation/impl/InstrumentationServiceImplTests$TestTargetClass",
-                        "instanceMethodNoArgs",
-                        List.of()
-                    )
-                ),
+                equalTo(new MethodKey("org/example/TestTargetClass", "instanceMethodNoArgs", List.of(), true)),
```

Comment:
Why is this passing? The "this" type is different from the type in the name (`InstrumentationServiceImplTests$TestTargetClass` vs `TestTargetClass`), aren't we checking for that?

Reply:
Which also means we are missing a test for when we have an internal class name -- since we use $ as a separator, one or more tests to show we don't get "confused" by it are important